From 87fc4690bb74c0b04b17b8bf0851dc06fc10a187 Mon Sep 17 00:00:00 2001
From: mactaggart <mactaggart@users.noreply.github.com>
Date: Fri, 13 Sep 2024 11:48:37 -0700
Subject: [PATCH] replace allowableValues in schema annotations with string
 array of values

use of a single command-delimited string results in a swagger spec where only
one value is specified. this is incorrect as generated code requires separate
values.
---
 .../apache/nifi/web/api/dto/AccessPolicySummaryDTO.java   | 2 +-
 .../org/apache/nifi/web/api/dto/AffectedComponentDTO.java | 6 +++---
 .../nifi/web/api/dto/ConfigVerificationResultDTO.java     | 2 +-
 .../java/org/apache/nifi/web/api/dto/ConnectableDTO.java  | 2 +-
 .../java/org/apache/nifi/web/api/dto/ConnectionDTO.java   | 6 +++---
 .../org/apache/nifi/web/api/dto/ControllerServiceDTO.java | 4 ++--
 .../api/dto/ControllerServiceReferencingComponentDTO.java | 2 +-
 .../org/apache/nifi/web/api/dto/FlowAnalysisRuleDTO.java  | 4 ++--
 .../apache/nifi/web/api/dto/FlowRegistryClientDTO.java    | 2 +-
 .../org/apache/nifi/web/api/dto/ParameterProviderDTO.java | 2 +-
 .../main/java/org/apache/nifi/web/api/dto/PortDTO.java    | 6 +++---
 .../java/org/apache/nifi/web/api/dto/ProcessGroupDTO.java | 8 ++++----
 .../org/apache/nifi/web/api/dto/ProcessorConfigDTO.java   | 2 +-
 .../java/org/apache/nifi/web/api/dto/ProcessorDTO.java    | 4 ++--
 .../nifi/web/api/dto/ProcessorRunStatusDetailsDTO.java    | 2 +-
 .../org/apache/nifi/web/api/dto/ReportingTaskDTO.java     | 4 ++--
 .../nifi/web/api/dto/VersionControlInformationDTO.java    | 2 +-
 .../org/apache/nifi/web/api/dto/VersionedFlowDTO.java     | 2 +-
 .../web/api/dto/provenance/lineage/LineageRequestDTO.java | 2 +-
 .../web/api/dto/provenance/lineage/ProvenanceNodeDTO.java | 2 +-
 .../nifi/web/api/dto/status/ComponentStatusDTO.java       | 4 ++--
 .../web/api/dto/status/ControllerServiceStatusDTO.java    | 2 +-
 .../web/api/dto/status/FlowAnalysisRuleStatusDTO.java     | 2 +-
 .../org/apache/nifi/web/api/dto/status/PortStatusDTO.java | 2 +-
 .../nifi/web/api/dto/status/PortStatusSnapshotDTO.java    | 2 +-
 .../web/api/dto/status/ProcessGroupStatusSnapshotDTO.java | 2 +-
 .../nifi/web/api/dto/status/ProcessorStatusDTO.java       | 2 +-
 .../web/api/dto/status/ProcessorStatusSnapshotDTO.java    | 4 ++--
 .../web/api/dto/status/RemoteProcessGroupStatusDTO.java   | 2 +-
 .../nifi/web/api/dto/status/ReportingTaskStatusDTO.java   | 2 +-
 .../web/api/entity/ActivateControllerServicesEntity.java  | 2 +-
 .../nifi/web/api/entity/AffectedComponentEntity.java      | 6 +++---
 .../org/apache/nifi/web/api/entity/ConnectionEntity.java  | 4 ++--
 .../web/api/entity/ControllerServiceRunStatusEntity.java  | 2 +-
 .../web/api/entity/FlowAnalysisRuleRunStatusEntity.java   | 2 +-
 .../apache/nifi/web/api/entity/FlowBreadcrumbEntity.java  | 2 +-
 .../apache/nifi/web/api/entity/PortRunStatusEntity.java   | 2 +-
 .../apache/nifi/web/api/entity/ProcessGroupEntity.java    | 4 ++--
 .../nifi/web/api/entity/ProcessorRunStatusEntity.java     | 2 +-
 .../nifi/web/api/entity/RemotePortRunStatusEntity.java    | 2 +-
 .../nifi/web/api/entity/ReplayLastEventRequestEntity.java | 2 +-
 .../web/api/entity/ReplayLastEventResponseEntity.java     | 2 +-
 .../nifi/web/api/entity/ReportingTaskRunStatusEntity.java | 2 +-
 .../nifi/web/api/entity/ScheduleComponentsEntity.java     | 2 +-
 .../UpdateControllerServiceReferenceRequestEntity.java    | 2 +-
 45 files changed, 64 insertions(+), 64 deletions(-)

diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/AccessPolicySummaryDTO.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/AccessPolicySummaryDTO.java
index 07da700c06..93db1d2773 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/AccessPolicySummaryDTO.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/AccessPolicySummaryDTO.java
@@ -36,7 +36,7 @@ public class AccessPolicySummaryDTO extends ComponentDTO {
      * @return The action associated with this access policy.
      */
     @Schema(description = "The action associated with this access policy.",
-            allowableValues = "read, write"
+            allowableValues = {"read", "write"}
     )
     public String getAction() {
         return action;
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/AffectedComponentDTO.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/AffectedComponentDTO.java
index 44f3024f88..d9ecfee34e 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/AffectedComponentDTO.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/AffectedComponentDTO.java
@@ -60,9 +60,9 @@ public class AffectedComponentDTO {
     }
 
     @Schema(description = "The type of this component",
-        allowableValues = COMPONENT_TYPE_PROCESSOR + "," + COMPONENT_TYPE_CONTROLLER_SERVICE + ", "
-            + COMPONENT_TYPE_INPUT_PORT + ", " + COMPONENT_TYPE_OUTPUT_PORT + ", "
-            + COMPONENT_TYPE_REMOTE_INPUT_PORT + ", " + COMPONENT_TYPE_REMOTE_OUTPUT_PORT + ", " + COMPONENT_TYPE_STATELESS_GROUP)
+        allowableValues = {COMPONENT_TYPE_PROCESSOR, COMPONENT_TYPE_CONTROLLER_SERVICE,
+            COMPONENT_TYPE_INPUT_PORT, COMPONENT_TYPE_OUTPUT_PORT,
+            COMPONENT_TYPE_REMOTE_INPUT_PORT, COMPONENT_TYPE_REMOTE_OUTPUT_PORT, COMPONENT_TYPE_STATELESS_GROUP })
     public String getReferenceType() {
         return referenceType;
     }
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ConfigVerificationResultDTO.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ConfigVerificationResultDTO.java
index 6ae28b2722..13841ca76f 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ConfigVerificationResultDTO.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ConfigVerificationResultDTO.java
@@ -24,7 +24,7 @@ public class ConfigVerificationResultDTO {
     private String verificationStepName;
     private String explanation;
 
-    @Schema(description = "The outcome of the verification", allowableValues = "SUCCESSFUL, FAILED, SKIPPED")
+    @Schema(description = "The outcome of the verification", allowableValues = {"SUCCESSFUL", "FAILED", "SKIPPED"})
     public String getOutcome() {
         return outcome;
     }
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ConnectableDTO.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ConnectableDTO.java
index fb58c2a344..553897b020 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ConnectableDTO.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ConnectableDTO.java
@@ -65,7 +65,7 @@ public class ConnectableDTO {
      */
     @Schema(description = "The type of component the connectable is.",
             requiredMode = Schema.RequiredMode.REQUIRED,
-            allowableValues = "PROCESSOR, REMOTE_INPUT_PORT, REMOTE_OUTPUT_PORT, INPUT_PORT, OUTPUT_PORT, FUNNEL"
+            allowableValues = {"PROCESSOR", "REMOTE_INPUT_PORT", "REMOTE_OUTPUT_PORT", "INPUT_PORT", "OUTPUT_PORT", "FUNNEL"}
     )
     public String getType() {
         return type;
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ConnectionDTO.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ConnectionDTO.java
index 22aea63a1c..ea655f4a27 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ConnectionDTO.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ConnectionDTO.java
@@ -228,7 +228,7 @@ public class ConnectionDTO extends ComponentDTO {
     }
 
     @Schema(description = "How to load balance the data in this Connection across the nodes in the cluster.",
-        allowableValues = "DO_NOT_LOAD_BALANCE, PARTITION_BY_ATTRIBUTE, ROUND_ROBIN, SINGLE_NODE")
+        allowableValues = {"DO_NOT_LOAD_BALANCE", "PARTITION_BY_ATTRIBUTE", "ROUND_ROBIN", "SINGLE_NODE"})
     public String getLoadBalanceStrategy() {
         return loadBalanceStrategy;
     }
@@ -247,7 +247,7 @@ public class ConnectionDTO extends ComponentDTO {
     }
 
     @Schema(description = "Whether or not data should be compressed when being transferred between nodes in the cluster.",
-        allowableValues = "DO_NOT_COMPRESS, COMPRESS_ATTRIBUTES_ONLY, COMPRESS_ATTRIBUTES_AND_CONTENT")
+        allowableValues = {"DO_NOT_COMPRESS", "COMPRESS_ATTRIBUTES_ONLY", "COMPRESS_ATTRIBUTES_AND_CONTENT"})
     public String getLoadBalanceCompression() {
         return loadBalanceCompression;
     }
@@ -258,7 +258,7 @@ public class ConnectionDTO extends ComponentDTO {
 
     @Schema(description = "The current status of the Connection's Load Balancing Activities. Status can indicate that Load Balancing is not configured for the connection, that Load Balancing " +
         "is configured but inactive (not currently transferring data to another node), or that Load Balancing is configured and actively transferring data to another node.",
-        allowableValues = LOAD_BALANCE_NOT_CONFIGURED + ", " + LOAD_BALANCE_INACTIVE + ", " + LOAD_BALANCE_ACTIVE,
+        allowableValues = {LOAD_BALANCE_NOT_CONFIGURED, LOAD_BALANCE_INACTIVE, LOAD_BALANCE_ACTIVE},
         accessMode = Schema.AccessMode.READ_ONLY)
     public String getLoadBalanceStatus() {
         return loadBalanceStatus;
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ControllerServiceDTO.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ControllerServiceDTO.java
index 7a7488a531..17a0bd1756 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ControllerServiceDTO.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ControllerServiceDTO.java
@@ -224,7 +224,7 @@ public class ControllerServiceDTO extends ComponentDTO {
      * @return The state of this controller service. Possible values are ENABLED, ENABLING, DISABLED, DISABLING
      */
     @Schema(description = "The state of the controller service.",
-            allowableValues = "ENABLED, ENABLING, DISABLED, DISABLING"
+            allowableValues = {"ENABLED", "ENABLING", "DISABLED", "DISABLING"}
     )
     public String getState() {
         return state;
@@ -332,7 +332,7 @@ public class ControllerServiceDTO extends ComponentDTO {
 
     @Schema(description = "Indicates whether the ControllerService is valid, invalid, or still in the process of validating (i.e., it is unknown whether or not the ControllerService is valid)",
         accessMode = Schema.AccessMode.READ_ONLY,
-        allowableValues = VALID + ", " + INVALID + ", " + VALIDATING)
+        allowableValues = {VALID, INVALID, VALIDATING})
     public String getValidationStatus() {
         return validationStatus;
     }
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ControllerServiceReferencingComponentDTO.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ControllerServiceReferencingComponentDTO.java
index 3cfd407fd2..01638672fa 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ControllerServiceReferencingComponentDTO.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ControllerServiceReferencingComponentDTO.java
@@ -118,7 +118,7 @@ public class ControllerServiceReferencingComponentDTO {
      * @return type of reference this is (Processor, ControllerService, ParameterProvider, or ReportingTask)
      */
     @Schema(description = "The type of reference this is.",
-            allowableValues = "Processor, ControllerService, ReportingTask, FlowRegistryClient"
+            allowableValues = {"Processor", "ControllerService", "ReportingTask", "FlowRegistryClient"}
     )
     public String getReferenceType() {
         return referenceType;
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/FlowAnalysisRuleDTO.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/FlowAnalysisRuleDTO.java
index 33e2058214..16be05f9cd 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/FlowAnalysisRuleDTO.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/FlowAnalysisRuleDTO.java
@@ -185,7 +185,7 @@ public class FlowAnalysisRuleDTO extends ComponentDTO {
      * @return current scheduling state of the flow analysis rule
      */
     @Schema(description = "The state of the flow analysis rule.",
-            allowableValues = "ENABLED, DISABLED"
+            allowableValues = {"ENABLED", "DISABLED"}
     )
     public String getState() {
         return state;
@@ -265,7 +265,7 @@ public class FlowAnalysisRuleDTO extends ComponentDTO {
 
     @Schema(description = "Indicates whether the Flow Analysis Rule is valid, invalid, or still in the process of validating (i.e., it is unknown whether or not the Flow Analysis Rule is valid)",
         accessMode = Schema.AccessMode.READ_ONLY,
-        allowableValues = VALID + ", " + INVALID + ", " + VALIDATING)
+        allowableValues = {VALID, INVALID, VALIDATING})
     public String getValidationStatus() {
         return validationStatus;
     }
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/FlowRegistryClientDTO.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/FlowRegistryClientDTO.java
index 5554466345..295763ffb8 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/FlowRegistryClientDTO.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/FlowRegistryClientDTO.java
@@ -213,7 +213,7 @@ public class FlowRegistryClientDTO {
 
     @Schema(description = "Indicates whether the Registry Client is valid, invalid, or still in the process of validating (i.e., it is unknown whether or not the Registry Client is valid)",
             accessMode = Schema.AccessMode.READ_ONLY,
-            allowableValues = VALID + ", " + INVALID + ", " + VALIDATING)
+            allowableValues = {VALID, INVALID, VALIDATING})
     public String getValidationStatus() {
         return validationStatus;
     }
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ParameterProviderDTO.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ParameterProviderDTO.java
index 635bf925d4..98084bc0a2 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ParameterProviderDTO.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ParameterProviderDTO.java
@@ -284,7 +284,7 @@ public class ParameterProviderDTO extends ComponentDTO {
 
     @Schema(description = "Indicates whether the Parameter Provider is valid, invalid, or still in the process of validating (i.e., it is unknown whether or not the Parameter Provider is valid)",
         accessMode = Schema.AccessMode.READ_ONLY,
-        allowableValues = VALID + ", " + INVALID + ", " + VALIDATING)
+        allowableValues = {VALID, INVALID, VALIDATING})
     public String getValidationStatus() {
         return validationStatus;
     }
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/PortDTO.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/PortDTO.java
index 8e2364aa95..cc143176a2 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/PortDTO.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/PortDTO.java
@@ -55,7 +55,7 @@ public class PortDTO extends ComponentDTO {
      * @return The state of this port. Possible states are 'RUNNING', 'STOPPED', and 'DISABLED'
      */
     @Schema(description = "The state of the port.",
-            allowableValues = "RUNNING, STOPPED, DISABLED"
+            allowableValues = {"RUNNING", "STOPPED", "DISABLED"}
     )
     public String getState() {
         return state;
@@ -71,7 +71,7 @@ public class PortDTO extends ComponentDTO {
      * @return The type of port
      */
     @Schema(description = "The type of port.",
-            allowableValues = "INPUT_PORT, OUTPUT_PORT"
+            allowableValues = {"INPUT_PORT", "OUTPUT_PORT"}
     )
     public String getType() {
         return type;
@@ -150,7 +150,7 @@ public class PortDTO extends ComponentDTO {
     }
 
     @Schema(description = "Specifies how the Port functions",
-        allowableValues = "STANDARD, FAILURE"
+        allowableValues = {"STANDARD", "FAILURE"}
     )
     public String getPortFunction() {
         return portFunction;
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ProcessGroupDTO.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ProcessGroupDTO.java
index b2b22d0f70..84ecbad8cb 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ProcessGroupDTO.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ProcessGroupDTO.java
@@ -335,7 +335,7 @@ public class ProcessGroupDTO extends ComponentDTO {
         this.parameterContext = parameterContext;
     }
 
-    @Schema(description = "The FlowFile Concurrency for this Process Group.", allowableValues = "UNBOUNDED, SINGLE_FLOWFILE_PER_NODE, SINGLE_BATCH_PER_NODE")
+    @Schema(description = "The FlowFile Concurrency for this Process Group.", allowableValues = {"UNBOUNDED", "SINGLE_FLOWFILE_PER_NODE", "SINGLE_BATCH_PER_NODE"})
     public String getFlowfileConcurrency() {
         return flowfileConcurrency;
     }
@@ -345,7 +345,7 @@ public class ProcessGroupDTO extends ComponentDTO {
     }
 
     @Schema(description = "The Outbound Policy that is used for determining how FlowFiles should be transferred out of the Process Group.",
-        allowableValues = "STREAM_WHEN_AVAILABLE, BATCH_OUTPUT")
+        allowableValues = {"STREAM_WHEN_AVAILABLE", "BATCH_OUTPUT"})
     public String getFlowfileOutboundPolicy() {
         return flowfileOutboundPolicy;
     }
@@ -391,7 +391,7 @@ public class ProcessGroupDTO extends ComponentDTO {
     }
 
     @Schema(description = "The Execution Engine that should be used to run the flow represented by this Process Group.",
-        allowableValues = "STATELESS, STANDARD, INHERITED")
+        allowableValues = {"STATELESS", "STANDARD", "INHERITED"})
     public String getExecutionEngine() {
         return executionEngine;
     }
@@ -401,7 +401,7 @@ public class ProcessGroupDTO extends ComponentDTO {
     }
 
     @Schema(description = "If the Process Group is configured to run in using the Stateless Engine, represents the current state. Otherwise, will be STOPPED.",
-            allowableValues = "STOPPED, RUNNING")
+            allowableValues = {"STOPPED", "RUNNING"})
     public String getStatelessGroupScheduledState() {
         return statelessGroupScheduledState;
     }
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ProcessorConfigDTO.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ProcessorConfigDTO.java
index 5f8bd5a7bf..f7be45edf4 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ProcessorConfigDTO.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ProcessorConfigDTO.java
@@ -332,7 +332,7 @@ public class ProcessorConfigDTO {
     }
 
     @Schema(description = "Determines whether the FlowFile should be penalized or the processor should be yielded between retries.",
-            allowableValues = "PENALIZE_FLOWFILE, YIELD_PROCESSOR"
+            allowableValues = {"PENALIZE_FLOWFILE", "YIELD_PROCESSOR"}
     )
     public String getBackoffMechanism() {
         return backoffMechanism;
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ProcessorDTO.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ProcessorDTO.java
index 1e9ce740b0..34a6511a5b 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ProcessorDTO.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ProcessorDTO.java
@@ -108,7 +108,7 @@ public class ProcessorDTO extends ComponentDTO {
      * @return The state of this processor. Possible states are 'RUNNING', 'STOPPED', and 'DISABLED'
      */
     @Schema(description = "The state of the processor",
-            allowableValues = "RUNNING, STOPPED, DISABLED"
+            allowableValues = {"RUNNING", "STOPPED", "DISABLED"}
     )
     public String getState() {
         return state;
@@ -296,7 +296,7 @@ public class ProcessorDTO extends ComponentDTO {
 
     @Schema(description = "Indicates whether the Processor is valid, invalid, or still in the process of validating (i.e., it is unknown whether or not the Processor is valid)",
         accessMode = Schema.AccessMode.READ_ONLY,
-        allowableValues = VALID + ", " + INVALID + ", " + VALIDATING)
+        allowableValues = {VALID, INVALID, VALIDATING})
     public String getValidationStatus() {
         return validationStatus;
     }
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ProcessorRunStatusDetailsDTO.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ProcessorRunStatusDetailsDTO.java
index da2161856a..aaf4aaff95 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ProcessorRunStatusDetailsDTO.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ProcessorRunStatusDetailsDTO.java
@@ -56,7 +56,7 @@ public class ProcessorRunStatusDetailsDTO {
     }
 
     @Schema(description = "The run status of the processor",
-        allowableValues = RUNNING + ", " + STOPPED + ", " + INVALID + ", " + VALIDATING + ", " + DISABLED
+        allowableValues = {RUNNING, STOPPED, INVALID, VALIDATING, DISABLED}
     )
     public String getRunStatus() {
         return runStatus;
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ReportingTaskDTO.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ReportingTaskDTO.java
index bec8284aaf..08ba132525 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ReportingTaskDTO.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/ReportingTaskDTO.java
@@ -210,7 +210,7 @@ public class ReportingTaskDTO extends ComponentDTO {
      * @return current scheduling state of the reporting task
      */
     @Schema(description = "The state of the reporting task.",
-            allowableValues = "RUNNING, STOPPED, DISABLED"
+            allowableValues = {"RUNNING", "STOPPED", "DISABLED"}
     )
     public String getState() {
         return state;
@@ -316,7 +316,7 @@ public class ReportingTaskDTO extends ComponentDTO {
 
     @Schema(description = "Indicates whether the Reporting Task is valid, invalid, or still in the process of validating (i.e., it is unknown whether or not the Reporting Task is valid)",
         accessMode = Schema.AccessMode.READ_ONLY,
-        allowableValues = VALID + ", " + INVALID + ", " + VALIDATING)
+        allowableValues = {VALID, INVALID, VALIDATING})
     public String getValidationStatus() {
         return validationStatus;
     }
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/VersionControlInformationDTO.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/VersionControlInformationDTO.java
index 3ccf1c4b06..efa508c4be 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/VersionControlInformationDTO.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/VersionControlInformationDTO.java
@@ -145,7 +145,7 @@ public class VersionControlInformationDTO {
 
     @Schema(accessMode = Schema.AccessMode.READ_ONLY,
         description = "The current state of the Process Group, as it relates to the Versioned Flow",
-        allowableValues = LOCALLY_MODIFIED + ", " + STALE + ", " + LOCALLY_MODIFIED_AND_STALE + ", " + UP_TO_DATE + ", " + SYNC_FAILURE)
+        allowableValues = {LOCALLY_MODIFIED, STALE, LOCALLY_MODIFIED_AND_STALE, UP_TO_DATE, SYNC_FAILURE})
     public String getState() {
         return state;
     }
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/VersionedFlowDTO.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/VersionedFlowDTO.java
index 14be45ca60..192fb4a1cd 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/VersionedFlowDTO.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/VersionedFlowDTO.java
@@ -98,7 +98,7 @@ public class VersionedFlowDTO {
         this.comments = comments;
     }
 
-    @Schema(description = "The action being performed", allowableValues = COMMIT_ACTION + ", " + FORCE_COMMIT_ACTION)
+    @Schema(description = "The action being performed", allowableValues = {COMMIT_ACTION, FORCE_COMMIT_ACTION})
     public String getAction() {
         return action;
     }
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/provenance/lineage/LineageRequestDTO.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/provenance/lineage/LineageRequestDTO.java
index 745e8ac017..8689811009 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/provenance/lineage/LineageRequestDTO.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/provenance/lineage/LineageRequestDTO.java
@@ -67,7 +67,7 @@ public class LineageRequestDTO {
      */
     @Schema(description = "The type of lineage request. PARENTS will return the lineage for the flowfiles that are parents of the specified event. CHILDREN will return the lineage "
                     + "for the flowfiles that are children of the specified event. FLOWFILE will return the lineage for the specified flowfile.",
-            allowableValues = "PARENTS, CHILDREN, and FLOWFILE"
+            allowableValues = {"PARENTS", "CHILDREN", "FLOWFILE"}
     )
     public LineageRequestType getLineageRequestType() {
         return lineageRequestType;
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/provenance/lineage/ProvenanceNodeDTO.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/provenance/lineage/ProvenanceNodeDTO.java
index 1fedc0fb6d..97bba83b08 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/provenance/lineage/ProvenanceNodeDTO.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/provenance/lineage/ProvenanceNodeDTO.java
@@ -109,7 +109,7 @@ public class ProvenanceNodeDTO {
      * @return type of node
      */
     @Schema(description = "The type of the node.",
-            allowableValues = "FLOWFILE, EVENT"
+            allowableValues = {"FLOWFILE", "EVENT"}
     )
     public String getType() {
         return type;
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/ComponentStatusDTO.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/ComponentStatusDTO.java
index f574c17cf7..f4e61763a6 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/ComponentStatusDTO.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/ComponentStatusDTO.java
@@ -44,7 +44,7 @@ public class ComponentStatusDTO {
      */
     @Schema(description = "The run status of this component",
             accessMode = Schema.AccessMode.READ_ONLY,
-            allowableValues = "ENABLED, ENABLING, DISABLED, DISABLING")
+            allowableValues = {"ENABLED", "ENABLING", "DISABLED", "DISABLING"})
     public String getRunStatus() {
         return runStatus;
     }
@@ -56,7 +56,7 @@ public class ComponentStatusDTO {
     @Schema(description = "Indicates whether the component is valid, invalid, or still in the process of validating" +
             " (i.e., it is unknown whether or not the component is valid)",
             accessMode = Schema.AccessMode.READ_ONLY,
-            allowableValues = VALID + ", " + INVALID + ", " + VALIDATING)
+            allowableValues = {VALID, INVALID, VALIDATING})
     public String getValidationStatus() {
         return validationStatus;
     }
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/ControllerServiceStatusDTO.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/ControllerServiceStatusDTO.java
index f6e97ac234..e436855a9d 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/ControllerServiceStatusDTO.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/ControllerServiceStatusDTO.java
@@ -29,7 +29,7 @@ public class ControllerServiceStatusDTO extends ComponentStatusDTO {
 
     @Schema(description = "The run status of this ControllerService",
             accessMode = Schema.AccessMode.READ_ONLY,
-            allowableValues = "ENABLED, ENABLING, DISABLED, DISABLING")
+            allowableValues = {"ENABLED", "ENABLING", "DISABLED", "DISABLING"})
     @Override
     public String getRunStatus() {
         return super.getRunStatus();
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/FlowAnalysisRuleStatusDTO.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/FlowAnalysisRuleStatusDTO.java
index a4897f6d54..6098bfda31 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/FlowAnalysisRuleStatusDTO.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/FlowAnalysisRuleStatusDTO.java
@@ -29,7 +29,7 @@ public class FlowAnalysisRuleStatusDTO extends ComponentStatusDTO {
 
     @Schema(description = "The run status of this FlowAnalysisRule",
         accessMode = Schema.AccessMode.READ_ONLY,
-        allowableValues = "ENABLED, DISABLED")
+        allowableValues = {"ENABLED", "DISABLED"})
     @Override
     public String getRunStatus() {
         return super.getRunStatus();
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/PortStatusDTO.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/PortStatusDTO.java
index 431e14f6d3..1ca4450424 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/PortStatusDTO.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/PortStatusDTO.java
@@ -77,7 +77,7 @@ public class PortStatusDTO {
 
 
     @Schema(description = "The run status of the port.",
-            allowableValues = "Running, Stopped, Validating, Disabled, Invalid")
+            allowableValues = {"Running", "Stopped", "Validating", "Disabled", "Invalid"})
     public String getRunStatus() {
         return runStatus;
     }
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/PortStatusSnapshotDTO.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/PortStatusSnapshotDTO.java
index dad3c74c93..76c4fdbe71 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/PortStatusSnapshotDTO.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/PortStatusSnapshotDTO.java
@@ -106,7 +106,7 @@ public class PortStatusSnapshotDTO implements Cloneable {
      */
     @Schema(
             description = "The run status of the port.",
-            allowableValues = "Running, Stopped, Validating, Disabled, Invalid"
+            allowableValues = {"Running", "Stopped", "Validating", "Disabled", "Invalid"}
     )
     public String getRunStatus() {
         return runStatus;
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/ProcessGroupStatusSnapshotDTO.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/ProcessGroupStatusSnapshotDTO.java
index 39593473c5..e19599404e 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/ProcessGroupStatusSnapshotDTO.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/ProcessGroupStatusSnapshotDTO.java
@@ -109,7 +109,7 @@ public class ProcessGroupStatusSnapshotDTO implements Cloneable {
 
     @Schema(description = "The current state of the Process Group, as it relates to the Versioned Flow",
             accessMode = Schema.AccessMode.READ_ONLY,
-            allowableValues = "LOCALLY_MODIFIED, STALE, LOCALLY_MODIFIED_AND_STALE, UP_TO_DATE, SYNC_FAILURE")
+            allowableValues = {"LOCALLY_MODIFIED", "STALE", "LOCALLY_MODIFIED_AND_STALE", "UP_TO_DATE", "SYNC_FAILURE"})
     public String getVersionedFlowState() {
         return versionedFlowState;
     }
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/ProcessorStatusDTO.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/ProcessorStatusDTO.java
index 36c3f27e43..24239357f9 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/ProcessorStatusDTO.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/ProcessorStatusDTO.java
@@ -78,7 +78,7 @@ public class ProcessorStatusDTO implements Cloneable {
     }
 
     @Schema(description = "The run status of the Processor",
-            allowableValues = "Running, Stopped, Validating, Disabled, Invalid")
+            allowableValues = {"Running", "Stopped", "Validating", "Disabled", "Invalid"})
     public String getRunStatus() {
         return runStatus;
     }
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/ProcessorStatusSnapshotDTO.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/ProcessorStatusSnapshotDTO.java
index 2e9d1ac245..8dc96dac6c 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/ProcessorStatusSnapshotDTO.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/ProcessorStatusSnapshotDTO.java
@@ -94,7 +94,7 @@ public class ProcessorStatusSnapshotDTO implements Cloneable {
      * @return run status of this processor
      */
     @Schema(description = "The state of the processor.",
-            allowableValues = "Running, Stopped, Validating, Disabled, Invalid"
+            allowableValues = {"Running", "Stopped", "Validating", "Disabled", "Invalid"}
     )
     public String getRunStatus() {
         return runStatus;
@@ -105,7 +105,7 @@ public class ProcessorStatusSnapshotDTO implements Cloneable {
     }
 
     @Schema(description = "Indicates the node where the process will execute.",
-            allowableValues = "ALL, PRIMARY"
+            allowableValues = {"ALL", "PRIMARY"}
     )
     public String getExecutionNode() {
         return executionNode;
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/RemoteProcessGroupStatusDTO.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/RemoteProcessGroupStatusDTO.java
index 0ebb33c742..777ddbc79c 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/RemoteProcessGroupStatusDTO.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/RemoteProcessGroupStatusDTO.java
@@ -123,7 +123,7 @@ public class RemoteProcessGroupStatusDTO {
     @Schema(description = "Indicates whether the component is valid, invalid, or still in the process of validating" +
             " (i.e., it is unknown whether or not the component is valid)",
             accessMode = Schema.AccessMode.READ_ONLY,
-            allowableValues = "VALID, INVALID, VALIDATING")
+            allowableValues = {"VALID", "INVALID", "VALIDATING"})
     public String getValidationStatus() {
         return validationStatus;
     }
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/ReportingTaskStatusDTO.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/ReportingTaskStatusDTO.java
index f7c3859fb3..cb6cb9a20c 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/ReportingTaskStatusDTO.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/ReportingTaskStatusDTO.java
@@ -33,7 +33,7 @@ public class ReportingTaskStatusDTO extends ComponentStatusDTO {
 
     @Schema(description = "The run status of this ReportingTask",
             accessMode = Schema.AccessMode.READ_ONLY,
-            allowableValues = "RUNNING, STOPPED, DISABLED")
+            allowableValues = {"RUNNING", "STOPPED", "DISABLED"})
     @Override
     public String getRunStatus() {
         return super.getRunStatus();
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ActivateControllerServicesEntity.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ActivateControllerServicesEntity.java
index 2a9f28966b..fe34a83b21 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ActivateControllerServicesEntity.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ActivateControllerServicesEntity.java
@@ -45,7 +45,7 @@ public class ActivateControllerServicesEntity extends Entity {
      * @return The desired state of the descendant components. Possible states are 'RUNNING' and 'STOPPED'
      */
     @Schema(description = "The desired state of the descendant components",
-        allowableValues = STATE_ENABLED + ", " + STATE_DISABLED)
+        allowableValues = {STATE_ENABLED, STATE_DISABLED})
     public String getState() {
         return state;
     }
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/AffectedComponentEntity.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/AffectedComponentEntity.java
index e4a8f145a6..14f871fbfe 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/AffectedComponentEntity.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/AffectedComponentEntity.java
@@ -57,9 +57,9 @@ public class AffectedComponentEntity extends ComponentEntity implements Permissi
     }
 
     @Schema(description = "The type of component referenced",
-        allowableValues = AffectedComponentDTO.COMPONENT_TYPE_PROCESSOR + "," + AffectedComponentDTO.COMPONENT_TYPE_CONTROLLER_SERVICE + ", "
-                + AffectedComponentDTO.COMPONENT_TYPE_INPUT_PORT + ", " + AffectedComponentDTO.COMPONENT_TYPE_OUTPUT_PORT + ", "
-                + AffectedComponentDTO.COMPONENT_TYPE_REMOTE_INPUT_PORT + ", " + AffectedComponentDTO.COMPONENT_TYPE_REMOTE_OUTPUT_PORT)
+        allowableValues = {AffectedComponentDTO.COMPONENT_TYPE_PROCESSOR, AffectedComponentDTO.COMPONENT_TYPE_CONTROLLER_SERVICE,
+                AffectedComponentDTO.COMPONENT_TYPE_INPUT_PORT, AffectedComponentDTO.COMPONENT_TYPE_OUTPUT_PORT,
+                AffectedComponentDTO.COMPONENT_TYPE_REMOTE_INPUT_PORT, AffectedComponentDTO.COMPONENT_TYPE_REMOTE_OUTPUT_PORT})
     public String getReferenceType() {
         return referenceType;
     }
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ConnectionEntity.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ConnectionEntity.java
index bfc222847b..0c95ade40b 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ConnectionEntity.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ConnectionEntity.java
@@ -149,7 +149,7 @@ public class ConnectionEntity extends ComponentEntity implements Permissible<Con
      */
     @Schema(description = "The type of component the source connectable is.",
             requiredMode = Schema.RequiredMode.REQUIRED,
-            allowableValues = "PROCESSOR, REMOTE_INPUT_PORT, REMOTE_OUTPUT_PORT, INPUT_PORT, OUTPUT_PORT, FUNNEL"
+            allowableValues = {"PROCESSOR", "REMOTE_INPUT_PORT", "REMOTE_OUTPUT_PORT", "INPUT_PORT", "OUTPUT_PORT, FUNNEL"}
     )
     public String getSourceType() {
         return sourceType;
@@ -177,7 +177,7 @@ public class ConnectionEntity extends ComponentEntity implements Permissible<Con
      */
     @Schema(description = "The type of component the destination connectable is.",
             requiredMode = Schema.RequiredMode.REQUIRED,
-            allowableValues = "PROCESSOR, REMOTE_INPUT_PORT, REMOTE_OUTPUT_PORT, INPUT_PORT, OUTPUT_PORT, FUNNEL"
+            allowableValues = {"PROCESSOR", "REMOTE_INPUT_PORT", "REMOTE_OUTPUT_PORT", "INPUT_PORT", "OUTPUT_PORT", "FUNNEL"}
     )
     public String getDestinationType() {
         return destinationType;
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ControllerServiceRunStatusEntity.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ControllerServiceRunStatusEntity.java
index 6110055bf1..42e6248559 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ControllerServiceRunStatusEntity.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ControllerServiceRunStatusEntity.java
@@ -39,7 +39,7 @@ public class ControllerServiceRunStatusEntity extends ComponentRunStatusEntity {
      * @return The run status
      */
     @Schema(description = "The run status of the ControllerService.",
-            allowableValues = "ENABLED, DISABLED"
+            allowableValues = {"ENABLED", "DISABLED"}
     )
     public String getState() {
         return super.getState();
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/FlowAnalysisRuleRunStatusEntity.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/FlowAnalysisRuleRunStatusEntity.java
index 25e6bb5857..6b00f5c6f0 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/FlowAnalysisRuleRunStatusEntity.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/FlowAnalysisRuleRunStatusEntity.java
@@ -38,7 +38,7 @@ public class FlowAnalysisRuleRunStatusEntity extends ComponentRunStatusEntity {
      * @return The state
      */
     @Schema(description = "The state of the FlowAnalysisRule.",
-            allowableValues = "ENABLED, DISABLED"
+            allowableValues = {"ENABLED", "DISABLED"}
     )
     public String getState() {
         return super.getState();
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/FlowBreadcrumbEntity.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/FlowBreadcrumbEntity.java
index d31b796b61..d81bf3eba3 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/FlowBreadcrumbEntity.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/FlowBreadcrumbEntity.java
@@ -96,7 +96,7 @@ public class FlowBreadcrumbEntity extends Entity {
 
     @Schema(
             description = "The current state of the Process Group, as it relates to the Versioned Flow",
-            allowableValues = "LOCALLY_MODIFIED, STALE, LOCALLY_MODIFIED_AND_STALE, UP_TO_DATE, SYNC_FAILURE",
+            allowableValues = {"LOCALLY_MODIFIED", "STALE", "LOCALLY_MODIFIED_AND_STALE", "UP_TO_DATE", "SYNC_FAILURE"},
             accessMode = Schema.AccessMode.READ_ONLY
     )
     public String getVersionedFlowState() {
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/PortRunStatusEntity.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/PortRunStatusEntity.java
index 106727dfa3..1973f7869b 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/PortRunStatusEntity.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/PortRunStatusEntity.java
@@ -38,7 +38,7 @@ public class PortRunStatusEntity extends ComponentRunStatusEntity {
      * @return The run status
      */
     @Schema(description = "The run status of the Port.",
-            allowableValues = "RUNNING, STOPPED, DISABLED"
+            allowableValues = {"RUNNING", "STOPPED", "DISABLED"}
     )
     public String getState() {
         return super.getState();
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ProcessGroupEntity.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ProcessGroupEntity.java
index cd5f452d40..22eb784d0d 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ProcessGroupEntity.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ProcessGroupEntity.java
@@ -255,7 +255,7 @@ public class ProcessGroupEntity extends ComponentEntity implements Permissible<P
 
     @Schema(
             description = "The current state of the Process Group, as it relates to the Versioned Flow",
-            allowableValues = "LOCALLY_MODIFIED, STALE, LOCALLY_MODIFIED_AND_STALE, UP_TO_DATE, SYNC_FAILURE",
+            allowableValues = {"LOCALLY_MODIFIED", "STALE", "LOCALLY_MODIFIED_AND_STALE", "UP_TO_DATE", "SYNC_FAILURE"},
             accessMode = Schema.AccessMode.READ_ONLY
     )
     public String getVersionedFlowState() {
@@ -321,7 +321,7 @@ public class ProcessGroupEntity extends ComponentEntity implements Permissible<P
     }
 
     @Schema(description = "Determines the process group update strategy",
-            allowableValues = "CURRENT_GROUP, CURRENT_GROUP_WITH_CHILDREN"
+            allowableValues = {"CURRENT_GROUP", "CURRENT_GROUP_WITH_CHILDREN"}
     )
     public String getProcessGroupUpdateStrategy() {
         return processGroupUpdateStrategy;
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ProcessorRunStatusEntity.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ProcessorRunStatusEntity.java
index 47d5a333db..7c6d7c9576 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ProcessorRunStatusEntity.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ProcessorRunStatusEntity.java
@@ -39,7 +39,7 @@ public class ProcessorRunStatusEntity extends ComponentRunStatusEntity {
      * Run status for this Processor.
      * @return The run status
      */
-    @Schema(description = "The run status of the Processor.", allowableValues = ALLOWABLE_VALUES)
+    @Schema(description = "The run status of the Processor.", allowableValues = {"RUNNING", "STOPPED", "DISABLED", "RUN_ONCE"})
     public String getState() {
         return super.getState();
     }
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/RemotePortRunStatusEntity.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/RemotePortRunStatusEntity.java
index abe972bf91..8ee5f8e4e5 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/RemotePortRunStatusEntity.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/RemotePortRunStatusEntity.java
@@ -38,7 +38,7 @@ public class RemotePortRunStatusEntity extends ComponentRunStatusEntity {
      * @return The run status
      */
     @Schema(description = "The run status of the RemotePort.",
-            allowableValues = "TRANSMITTING, STOPPED"
+            allowableValues = {"TRANSMITTING", "STOPPED"}
     )
     public String getState() {
         return super.getState();
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ReplayLastEventRequestEntity.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ReplayLastEventRequestEntity.java
index d2bc325b2d..0d63949c97 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ReplayLastEventRequestEntity.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ReplayLastEventRequestEntity.java
@@ -36,7 +36,7 @@ public class ReplayLastEventRequestEntity extends Entity {
     }
 
     @Schema(description = "Which nodes are to replay their last provenance event.",
-        allowableValues = "ALL, PRIMARY"
+        allowableValues = {"ALL", "PRIMARY"}
     )
     public String getNodes() {
         return nodes;
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ReplayLastEventResponseEntity.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ReplayLastEventResponseEntity.java
index dc694a749f..1f9aa34eaf 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ReplayLastEventResponseEntity.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ReplayLastEventResponseEntity.java
@@ -39,7 +39,7 @@ public class ReplayLastEventResponseEntity extends Entity {
     }
 
     @Schema(description = "Which nodes were requested to replay their last provenance event.",
-        allowableValues = "ALL, PRIMARY"
+        allowableValues = {"ALL", "PRIMARY"}
     )
     public String getNodes() {
         return nodes;
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ReportingTaskRunStatusEntity.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ReportingTaskRunStatusEntity.java
index db9a70b989..9e94f3c330 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ReportingTaskRunStatusEntity.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ReportingTaskRunStatusEntity.java
@@ -38,7 +38,7 @@ public class ReportingTaskRunStatusEntity extends ComponentRunStatusEntity {
      * @return The run status
      */
     @Schema(description = "The run status of the ReportingTask.",
-            allowableValues = "RUNNING, STOPPED"
+            allowableValues = {"RUNNING", "STOPPED"}
     )
     public String getState() {
         return super.getState();
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ScheduleComponentsEntity.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ScheduleComponentsEntity.java
index 9ec1092cd6..c33141ea54 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ScheduleComponentsEntity.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/ScheduleComponentsEntity.java
@@ -54,7 +54,7 @@ public class ScheduleComponentsEntity extends Entity {
      * @return The desired state of the descendant components. Possible states are 'RUNNING', 'STOPPED', 'ENABLED', and 'DISABLED'
      */
     @Schema(description = "The desired state of the descendant components",
-        allowableValues = STATE_RUNNING + ", " + STATE_STOPPED + ", " + STATE_ENABLED + ", " + STATE_DISABLED
+        allowableValues = {STATE_RUNNING, STATE_STOPPED, STATE_ENABLED, STATE_DISABLED}
     )
     public String getState() {
         return state;
diff --git a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/UpdateControllerServiceReferenceRequestEntity.java b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/UpdateControllerServiceReferenceRequestEntity.java
index 181e793e28..2da456622f 100644
--- a/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/UpdateControllerServiceReferenceRequestEntity.java
+++ b/nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/entity/UpdateControllerServiceReferenceRequestEntity.java
@@ -45,7 +45,7 @@ public class UpdateControllerServiceReferenceRequestEntity extends Entity {
     }
 
     @Schema(description = "The new state of the references for the controller service.",
-        allowableValues = "ENABLED, DISABLED, RUNNING, STOPPED"
+        allowableValues = {"ENABLED", "DISABLED", "RUNNING", "STOPPED"}
     )
     public String getState() {
         return state;
-- 
2.39.3 (Apple Git-146)

